source/Unsafe/CoNat-Equiv.lagda

```diff
@@ -0,0 +1,138 @@
+Alice Laroche, 4th of December 2024
+
+We show that the type of conaturals defined by coinduction is equivalent to the
+type of conaturals defined as generic convergent sequences when assuming funext
+and that bisimilarity is equality.
+
+\begin{code}
+
+{-# OPTIONS --guardedness #-} 
+
+module Unsafe.CoNat-Equiv where
+
+open import CoNaturals.Type
+open import MLTT.Spartan
+open import MLTT.Two-Properties
+open import UF.Base
+open import UF.Equiv
+open import UF.FunExt
+open import TypeTopology.Cantor
+
+\end{code}
+
+This implementation of CoNat comes from Cubical, the bisimilarity relation can
+be proven to be equivalent to equality, but not in classical Agda
+
+\begin{code}
+
+CoNat' : Set
+record CoNat : Set
+
+CoNat' = 𝟙 + CoNat
+record CoNat where
+ constructor conat
+ coinductive
+ field
+  force : 𝟙 + CoNat
+
+open CoNat public
+
+pattern cozero = inl ⋆
+pattern cosuc n = inr n
+
+record _＝C_ (x y : CoNat) : Set
+data _＝C'_ (x y : CoNat') : Set
+_＝C''_ : CoNat' → CoNat' → Set
+
+cozero  ＝C'' cozero  = 𝟙
+cozero  ＝C'' cosuc y = 𝟘
+cosuc x ＝C'' cozero  = 𝟘
+cosuc x ＝C'' cosuc y = x ＝C y
+
+data _＝C'_  x y where
+    con : x ＝C'' y → x ＝C' y
+    
+record _＝C_ x y where
```

Comment:
Maybe a few more comments would be in order, in the spirit of `TypeTopology`, which tried (often, but not always) to explain what's going on. I suppose this is the bisimilarity relation aluded to above.